PixaPicker/ViewController.swift

```diff
@@ -7,16 +7,33 @@
 //
 
 import UIKit
-
 import SDWebImage
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, UISearchResultsUpdating, UISearchBarDelegate {
+    
+    let apiKey = "key=13683470-874d69ddffa828cbb82551e32" //API key for my Pixabay account
     @IBOutlet weak var picsCollection: UICollectionView!
-    @IBOutlet weak var searchBar: UISearchBar!
+    
+    func updateSearchResults(for searchController: UISearchController) {
+        let searchText = searchController.searchBar.text!
+        let searchTextWithoutSpaces = searchText.replacingOccurrences(of: " ", with: "+")
+        let urlWithAPI = "https://pixabay.com/api/?" + apiKey
+        let urlWithSeachText = urlWithAPI + "&q=" + searchTextWithoutSpaces
+        let finalURL = urlWithSeachText + "&image_type=photo"
+        print(finalURL) //FOR TESTING ONLY
```

Comment:
The logic for constructing your `URL` should be moved into its own file so it can be reused easily elsewhere.  

What I like to do in these cases is create a `URL+Extensions.swift` or named similar file, create an `extension URL`, then create a `static func pixabaySearchURL(with text: String)` and add all the logic there.

Also, look into `URLComponents` for URL creating.

Reply:
Btw, I like your variable names, they are SUPER clear to read.